Tests/FluentAssertions.Specs/Xml/XDocumentAssertionSpecs.cs

```diff
@@ -1231,6 +1231,221 @@ public void When_asserting_a_document_has_an_element_with_a_null_xname_it_should
         }
     }
 
+    public class HaveElementWithValue
+    {
+        [Fact]
+        public void Throws_when_null()
+        {
+            // Arrange
+            XDocument element = null;
+
+            // Act
+            Action act = () => element.Should().HaveElementWithValue("child", "b");
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*child*b*element itself is <null>*");
+        }
+
+        [Fact]
+        public void Has_element_with_specified_value()
+        {
+            // Arrange
+            var element = XDocument.Parse(
+                """
+                <parent>
+                    <child>a</child>
+                    <child>b</child>
+                </parent>
+                """);
+
+            // Act / Assert
+            element.Should().HaveElementWithValue("child", "b");
+        }
+
+        [Fact]
+        public void Throws_when_element_not_found()
+        {
+            // Arrange
+            var element = XDocument.Parse(
+                """
+                <parent>
+                    <child>a</child>
+                    <child>b</child>
+                </parent>
+                """);
+
+            // Act
+            Action act = () => element.Should().HaveElementWithValue("grandchild", "f");
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*grandchild*f*element*isn't found*");
+        }
+
+        [Fact]
+        public void Throws_when_element_found_but_value_does_not_match()
+        {
+            // Arrange
+            var element = XDocument.Parse(
+                """
+                <parent>
+                    <child>a</child>
+                    <child>b</child>
+                </parent>
+                """);
+
+            // Act
+            Action act = () => element.Should().HaveElementWithValue("child", "c");
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("*child*c*element*does not have such a value*");
+        }
+
+        [Fact]
+        public void Throws_when_expected_element_is_null()
+        {
+            // Arrange
+            var element = XDocument.Parse(
+                """
+                <parent>
+                    <child>a</child>
+                    <child>b</child>
+                </parent>
+                """);
+
+            // Act
+            Action act = () => element.Should().HaveElementWithValue(null, "a");
+
+            // Assert
+            act.Should().Throw<ArgumentNullException>().WithMessage("*expectedElement*");
+        }
+
+        [Fact]
+        public void Throws_when_expected_element_is_null_with_ns()
```

Comment:
🤔I don't understand this test.

Reply:
Just checking the path with an added `XNamespace`